drivers/starburst/src/metabase/driver/implementation/execute.clj

```diff
@@ -158,23 +175,37 @@
         (catch Throwable e
           (log/debug e (trs "Error setting prepared statement fetch direction to FETCH_FORWARD"))))
       (sql-jdbc.execute/set-parameters! driver stmt params)
-      stmt
+      (proxy [java.sql.PreparedStatement] []
```

Comment:
If I understand this correctly, you are implementing the interface `PreparedStatement`, and making sure the method `executeQuery` removes the impersonation after the statement is executed.

Where in the code will this `executeQuery` actually be called? Why do we need to remove the impersonate there?

Reply:
`executeQuery` is called by Metabase. And we need to remove the impersonation because it's set at the connection level, so we don't want it to "leak" to another query

Reply:
I see, so it's set per connection?